### PR TITLE
Edit Product: add inline error UI for SKU local validation error

### PR DESCRIPTION
**Pull request by jaclync**
Part of #1652 

When the user entered a SKU value that is the same as another local Product in storage, this PR replaces the error alert with an inline error UI that highlights the section and row where the error occurs.

## Changes

- Created `ErrorSectionHeaderView` and its xib for a section header view to show an error state
- Added state to `TitleAndTextFieldTableViewCell.ViewModel` with default to `normal` state
- In `ProductInventorySettingsViewController`, registered the new `ErrorSectionHeaderView` header view. Added an optional `errorTitle` to the `Section` struct, and showed the error section header in its `UITableViewDelegate` implementation
  - I learned from trials and errors that just implementing `heightForHeaderInSection` that returns `UITableView.automaticDimension` wasn't enough. `estimatedHeightForHeaderInSection` was also required, otherwise `viewForHeaderInSection` for **the first section** (only the first section) wouldn't be called 😱 
- In `ProductInventorySettingsViewController`, when receiving a SKU error:
  - Stores the error
  - Reloads the table view section view models
  - Reloads the table view
    - The error state section header is shown by `ErrorSectionHeaderView` via the section view model with `errorTitle`
    - The error state in `TitleAndTextFieldTableViewCell` is shown by checking the internal `error` property and updating its cell view model's state

## Testing

- Launch the app
- Go to Products tab
- Tap on a simple Product
- Tap "Edit"
- Tap the inventory settings row
- Enter a SKU that is used by another existing Product in the storage (already loaded on the Products tab)
- Tap "Done" --> an error state is shown
- Edit the SKU to a different value that hasn't been taken
- Tap "Done" --> the SKU is saved locally but not yet published
- Tap "Done" on the Edit Product main screen --> the SKU should be saved remotely after

## Example screenshots

@kyleaparker I made a section header view for the error above the SKU section, so we can potentially reuse this UI for an error in a section (because there is already a top padding in the grouped style table view, I made the error label with 0px top margin and 22px bottom margin)

Dark | Light
-- | --
Simulator Screen Shot - iPhone 8 - 2020-01-03 at 20 52 08 | Simulator Screen Shot - iPhone 8 - 2020-01-03 at 20 52 18

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/ReusableViews/ErrorSectionHeaderView.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    @IBOutlet private weak var titleLabel: UILabel!
+
+    // MARK: - Overridden Methods
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        tintColor = .clear
+
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/ReusableViews/ErrorSectionHeaderView.swift]: Is this a desired space?

Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/ReusableViews/ErrorSectionHeaderView.swift]: not intentional! removed in https://github.com/woocommerce/woocommerce-ios/pull/1695/commits/958af2770f9ae50478664afd9ac76da16cf17269

**Comment by jaclync** [comment]
> Can the VoiceOver support changes would be made in this PR?

thanks for suggesting the VoiceOver support for error validation! I updated in https://github.com/woocommerce/woocommerce-ios/pull/1695/commits/9a6bff27f4e05453add3a93673fedebdf4a88ab5 - I tested with the screenshot below when the error is shown, see if you'd like to test it?

**Comment by jaclync** [comment]
> The notification for VoiceOver shouldn't land on the SKU textfield after an error. It should land on the inline label that says "SKU already in use by another product". You might be able to use tableview: shouldHighlightRow: to achieve this.

I see, that makes sense. I updated to post the accessibility notification with the error header view https://github.com/woocommerce/woocommerce-ios/pull/1695/commits/2f242f16fe098da166a313af05ec9f782bf9bb55 and tested it on a physical device. Now it lands on the error header, then I can swipe forward to the SKU text field:

**Comment by mindgraffiti** [comment]
Works perfectly 💯 Thanks Jaclyn!

**Pull request merged**